src/net.cpp

```diff
@@ -2418,7 +2418,7 @@ bool CConnman::Bind(const CService &addr, unsigned int flags, NetPermissionFlags
         return false;
     }
 
-    if (addr.IsRoutable() && fDiscover && !(flags & BF_DONT_ADVERTISE) && !(permissions & PF_NOBAN)) {
+    if (addr.IsRoutable() && fDiscover && !(flags & BF_DONT_ADVERTISE) && !NetPermissions::HasFlag(permissions, PF_NOBAN)) {
```

Comment:
sure, it's done in https://github.com/bitcoin/bitcoin/pull/21506/commits/873bbe6e46b8561acea365cb0fffa226fd891826#diff-00021eed586a482abdb09d6cdada1d90115abe988a91421851960e26658bed02R2413 but may as well update it here since I'm retouching both pulls